apps/test/unit/code-studio/utilsTest.js

```diff
@@ -34,15 +34,23 @@ describe('utils', () => {
 
   describe('updateQueryParam', () => {
     let pushStateOrig = window.history.pushState;
+    let replaceStateOrig = window.history.replaceState;
```

Comment:
I spent a little bit of time trying to do this. It might just be that I'm not great at sinon. I think part of the challenge is that we want to both override the behavior (stub) and check how it was called (spy). It seemed like mock might be the way to do this, but after spending some time on it, I was failing to get things working, and went with the quick and easy fix instead.